### PR TITLE
Changes for websocket entities

### PR DESCRIPTION
**Pull request by jijohn14**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [X] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by solankisamir** [comment]
>               }

missing 
` {
                "value": "webscoket",
                "description": "Imports the Soap API having a SOAP front end.",
                "name": "SoapPassThrough"
              
}`

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json:416 in f6d85d7. [](commit_id = f6d85d78a51a7ba9083f33c0a89074f7ce2ca185, deletion_comment = False)

**Comment by solankisamir** [comment]
>               "modelAsString": false

let flip this to `modelAsString`: true

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json:282 in f6d85d7. [](commit_id = f6d85d78a51a7ba9083f33c0a89074f7ce2ca185, deletion_comment = False)

**Comment by solankisamir** [comment]
>               "modelAsString": false

lets flip this to `modelAsString`:true

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json:748 in f6d85d7. [](commit_id = f6d85d78a51a7ba9083f33c0a89074f7ce2ca185, deletion_comment = False)

**Comment by solankisamir** [comment]
>               "modelAsString": false

flip to modelAsString `true`

---
Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json:803 in f6d85d7. [](commit_id = f6d85d78a51a7ba9083f33c0a89074f7ce2ca185, deletion_comment = False)

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json [hunk 1]
```diff
@@ -788,7 +794,9 @@
             "type": "string",
             "enum": [
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json:795]: will be great to have an example here on how to create a websocket based api
`[https://github.com/Azure/azure-rest-api-specs/blob/3f6e9241c599656f29b56d638798f31a6df9338f/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-01-01-preview/apimapis.json#L226`](https://github.com/Azure/azure-rest-api-specs/blob/3f6e9241c599656f29b56d638798f31a6df9338f/specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-01-01-preview/apimapis.json#L226`)

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json [hunk 2]
```diff
@@ -391,10 +393,11 @@
         },
         "apiType": {
           "type": "string",
-          "description": "Type of Api to create. \n * `http` creates a SOAP to REST API \n * `soap` creates a SOAP pass-through API .",
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json:394]: why remove the description. Instead, we can add details about WebSocket here.

**Review by solankisamir — CHANGES_REQUESTED** [review]
- request to add an example, which will run validation
- request to fix the CI error.

**Comment by jijohn14** [comment]
> n": "Imports the W

This change for soap api and not applicable for websocket

**Comment by jijohn14** [comment]
> > ```
> >           }
> > ```
> 
> missing
> ` {
> "value": "webscoket",
> "description": "Imports the Websocket API.",
> "name": "WebSocket"
> 
> }`
> 
> https://github.com/Azure/azure-rest-api-specs/pull/11912/checks?check_run_id=1942814117
> 
> Refers to: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/definitions.json:416 in [f6d85d7](https://github.com/Azure/azure-rest-api-specs/commit/f6d85d78a51a7ba9083f33c0a89074f7ce2ca185). [](commit_id = [f6d85d7](https://github.com/Azure/azure-rest-api-specs/commit/f6d85d78a51a7ba9083f33c0a89074f7ce2ca185), deletion_comment = False)

Applicable for soap api and not for websocket

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-01-01-preview/examples/ApiManagementCreateWebsocketApi.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    "201": {
+      "body": {
+        "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/apis/apiid9419",
+        "type": "Microsoft.ApiManagement/service/apis",
+        "name": "apiid9419",
+        "properties": {
+          "displayName": "apiname1463",
+          "apiRevision": "1",
+          "description": "apidescription5200",
+          "serviceUrl": "wss://echo.websocket.org",
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-01-01-preview/examples/ApiManagementCreateWebsocketApi.json:32]: reference this example from `apimapis.json` file

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-01-01-preview/apimapis.json [hunk 4]
```diff
@@ -242,6 +242,9 @@
           "ApiManagementCreateApi": {
             "$ref": "./examples/ApiManagementCreateApi.json"
           },
+          "ApiManagementCreateWebSocketApi": {
+            "$ref": "./examples/ApiManagementCreateWebSocketApi.json"
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-01-01-preview/apimapis.json:246]: >ApiManagementCreateWebSocketApi [](start = 32, length = 31)

need to fix the file name `ApiManagementCreateWebsocketApi`

**Review by solankisamir — APPROVED** [review]

**Comment by solankisamir** [comment]
@ArcturusZhang can you please review and merge this? We are working on a deadline for the feature announcement.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**